### PR TITLE
Assorted refinements

### PR DESCRIPTION
**Pull request by briskt**
### Changed
- Restrict files included in the Docker image
- Updated docker compose
- Restrict CI workflow to non-Terraform files
- Reduced memory to 128 MB
- Reduced timeout to 30 seconds

### Added
- Added a CODEOWNERS file

### CODE REVIEW

File: serverless.yml [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
+
 provider:
   name: aws
   runtime: provided.al2
   stage: prod
-  timeout: 30
+  timeout: 300
   versionFunctions: false
-  memorySize: 512
+  memorySize: 128
```
Comment (hobbitronics) [inline serverless.yml:11]: I am guessing we weren't coming anywhere close to 512?

**Review by hobbitronics — APPROVED** [review]
Nice refinements!

File: serverless.yml (hunk 1 above)
Comment (briskt) [reply serverless.yml:11]: Nope. I tried to figure out how much, but our biggest personnel-sync Lambda is configured for 128MB and it's been working for years.

**Comment by briskt** [comment]
> Nice refinements!

Thanks. :)

**Comment by forevermatt** [comment]
@briskt The description mentions "Reduced timeout to 30 seconds," but it looks like the code change was actually increasing it from 30 seconds to 5 minutes. I'm fine with either, but just wanted to verify which you intended / to ask about the discrepancy. Thanks for your work on this!

**Comment by briskt** [comment]
> @briskt The description mentions "Reduced timeout to 30 seconds," but it looks like the code change was actually increasing it from 30 seconds to 5 minutes. I'm fine with either, but just wanted to verify which you intended / to ask about the discrepancy. Thanks for your work on this!

Haha, I don't even remember why I changed that or intended to accomplish. These were from stashes or commits I had sitting on my local clone. I think that when I wrote the PR description I misread the change.

**Pull request merged**